Dev

clemalex06 opened the pull request.

Commit:
Creating First Blazor wasm application

Commit:
adding sql lite database and the summary

Commit:
Asding somes notes for dev part

Commit:
Create new solution with JWT authent first implem

Commit:
continue Registration form

Commit:
Continue jwt authent

Commit:
remove unsued call to chek if user is connected

Commit:
small refactoring for User registration

Commit:
Last Refacto on AuthController

Commit:
refactor UserService

Commit:
Using Newtonsoft.Json to store Users datas

Commit:
Rename UserService

Commit:
Add UserSqlService

Commit:
Refacto on Razor components

Commit:
First Implementation MarkDown controller

Commit:
Implemntation In progress

Commit:
first implementation markdig on front

clemalex06 committed:
Merge pull request #1 from clemalex06/MardownImplementation

Mardown implementation

Commit:
first sql lite implementation

Commit:
cleaning on Users first implementation

Commit:
make some clean

Commit:
Implement sqlite connections for documents

Commit:
add cssclass for text-area

Commit:
small improvment on front part

Commit:
Implementing small changes on front part

Commit:
removed useless package

Commit:
Implement Post method in MarkDownController

Commit:
finishing implementing update call

Commit:
finishing implementing first right management on mardown front component

Commit:
Implmenting Uspert method in DocumentService

Commit:
FinishImplementation for update

clemalex06 committed:
Merge pull request #2 from clemalex06/SqlLiteImplementation

Sql lite implementation

Commit:
finsih implementing unit tests

Commit:
unit tests, some refactoring :

Commit:
unit tests for services

Commit:
adding some notes to publish git hub peer reviews

Commit:
unit tests for AuthController

Commit:
finish unit tests

clemalex06 committed:
Merge pull request #3 from clemalex06/UnitTests

finsih implementing unit tests

Commit:
some clean up in the front part

Commit:
continue refactoring

Commit:
refactor front OK

Commit:
small refactoring

Commit:
small refactoring

Commit:
adding spinner loading the datas

Commit:
adding spinner on register and login form

Commit:
load ReadMe in about

Commit:
small refactoring

Commit:
add spinner at the start of the app

Commit:
adding unit tests

Commit:
wip

Commit:
create list

Commit:
wip

Commit:
wip

Commit:
wip

Commit:
small refactoring

Commit:
finishing implemnting create one

Commit:
Fix unit tests

Commit:
remove useless code

Commit:
managing error and success messages

Commit:
modify time logged

clemalex06 committed:
Merge pull request #4 from clemalex06/LastRefacto

Last refacto

Commit:
Merge branch 'dev' into addListOfMarkdown

Commit:
add README template

clemalex06 committed:
Merge pull request #5 from clemalex06/addListOfMarkdown

Add list of markdown

Commit:
Merge branch 'dev' into readme

Commit:
starting writing README.md

Commit:
wip

Commit:
implement readme

Commit:
last commit

clemalex06 committed:
Merge pull request #7 from clemalex06/readme

Readme

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Services/Users/UserService.cs:
```diff
@@ -0,0 +1,37 @@
+
+        public async Task<User?> AuthenticateUser(string? email, string? password)
+        {
+            var user = await Ctx.Users.SingleOrDefaultAsync(u => u.Email == email && u.Password == password);
```
You didn't check any entry parameter send in AuthenticateUser ?

clemalex06 replied:
I added this check on the following commit : 
https://github.com/clemalex06/BlazorMarkDown/pull/8/commits/614a92a5082bffb7dba45a98b6d1eef6dfdb552e

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Services/Users/IUserService.cs:
```diff
@@ -0,0 +1,10 @@
+{
+    public interface IUserService
+    {
+        Task<User?> AuthenticateUser(string? email, string? password);
```
**Remark 1**
It's a good pratice in .NET to suffix methods which return a Task with the Async suffix

More information below :
https://docs.microsoft.com/en-us/dotnet/standard/asynchronous-programming-patterns/task-based-asynchronous-pattern-tap?redirectedfrom=MSDN

jgiacomini replied:
**Remark 2**
A good pratice in ASP.NET Core is to add as latest parameter a CancellationToken.
This cancellation token will be filled automatically by ASP.NET Core if the TCP connection is closed before the end of computation on server side, you will be able to cancel all treament on server side and avoid to compute useless data.

More information below :
https://andrewlock.net/using-cancellationtokens-in-asp-net-core-mvc-controllers/

clemalex06 replied:
I could fix it on the last commit : 
https://github.com/clemalex06/BlazorMarkDown/pull/8/commits/614a92a5082bffb7dba45a98b6d1eef6dfdb552e

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Client/BlazorMarkDownAppJwt.Client.csproj:
```diff
@@ -0,0 +1,20 @@
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="6.0.5" />
+    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="6.0.5" PrivateAssets="all" />
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
```
Why do you need Newtownsoft.Json?

clemalex06 replied:
This is the nuget I am used to using to Serialize / Deserialize C# objects in json contents.

It's actually used on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Client/Components/MarkDownComponent.razor

![image](https://user-images.githubusercontent.com/9862193/189061975-e11ed8ca-39c4-4572-b7a2-5d214a132e3e.png)


I saw I can remove this package and use the System.Text.Json instead : 

https://docs.microsoft.com/en-gb/dotnet/standard/serialization/system-text-json-how-to?pivots=dotnet-6-0

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Client/Components/MarkDownComponent.razor:
```diff
@@ -0,0 +1,233 @@
+
+	private async Task<string> GetJWT()
+	{
+		var jwt = await jsr.InvokeAsync<string>("localStorage.getItem", "user").ConfigureAwait(false);
```
JFI the lib below allow you to have access to local storage  in C#
https://github.com/Blazored/LocalStorage

clemalex06 replied:
Ok I could have a look on the documentation, but I let my actual code like it is actually.

I wanted to go deeper with you to understand better the benefits to use it instead of IJSRuntime instanciation.

StillShedy replied:
this nuget use JSRuntime under the hood) but it just give u easy usage of localstorage
https://github.com/Blazored/LocalStorage/blob/main/src/Blazored.LocalStorage/BrowserStorageProvider.cs

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Client/Components/MarkDownComponent.razor:
```diff
@@ -0,0 +1,233 @@
+		successMessage = string.Empty;
+		errorMessage = string.Empty;
+		try
+		{
```
It's a good pratice to have a dedicated class to handle all communication between client an server.
It's seem very hardcoded.

clemalex06 replied:
Yes I agree with you that this part needs some refactoring. Unfortunatelly I didn't implement here.

Small question : where this dedicated class should be put ? I guess on the client project but I prefer to ask just to be sure

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Services/Users/UserService.cs:
```diff
@@ -0,0 +1,37 @@
+            Ctx = ctx;
+        }
+
+        public async Task<User?> AddUser(User addedUser)
```
**Remark 1**

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Services/MarkDowns/IDocumentService.cs:
```diff
@@ -0,0 +1,17 @@
+{
+    public interface IDocumentService
+    {
+        Task<Document?> GetDocument(long id);
```
**Remark 1** **Remark 2**

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Services/MarkDowns/IDocumentService.cs:
```diff
@@ -0,0 +1,17 @@
+    {
+        Task<Document?> GetDocument(long id);
+
+        Task<List<Document>> GetAllDocuments();
```
**Remark 1** **Remark 2**

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Services/MarkDowns/IDocumentService.cs:
```diff
@@ -0,0 +1,17 @@
+
+        Task<List<Document>> GetAllDocuments();
+
+        Task<Document?> InsertDocument(string markDown);
```
**Remark 1** **Remark 2**

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Services/MarkDowns/IDocumentService.cs:
```diff
@@ -0,0 +1,17 @@
+
+        Task<Document?> UpdateDocument(long id, string markDown);
+
+        Task<Document?> GetReadMeDocument();
```
**Remark 1** **Remark 2**

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Services/MarkDowns/IDocumentService.cs:
```diff
@@ -0,0 +1,17 @@
+
+        Task<Document?> InsertDocument(string markDown);
+
+        Task<Document?> UpdateDocument(long id, string markDown);
```
**Remark 1** **Remark 2**

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Services/MarkDowns/DocumentService.cs:
```diff
@@ -0,0 +1,71 @@
+        }
+
+		public async Task<Document?> UpdateDocument(long id, string markDown)
+		{
```
Wrong identation

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Pages/Error.cshtml.cs:
```diff
@@ -0,0 +1,27 @@
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using System.Diagnostics;
+
+namespace BlazorMarkDownAppJwt.Server.Pages
```
**Remark 4**  Do you need this page?

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Pages/Error.cshtml:
```diff
@@ -0,0 +1,42 @@
+﻿@page
```
**Remark 4**

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Controllers/MarkDownController.cs:
```diff
@@ -0,0 +1,176 @@
+            }
+            catch (Exception ex)
+            {
+                return StatusCode((int)HttpStatusCode.InternalServerError, ex);
```
**Remark 3**

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Controllers/MarkDownController.cs:
```diff
@@ -0,0 +1,176 @@
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(MarkDownModel))]
+        public async Task<IActionResult> Post([FromBody] MarkDownModel markDownModel)
+        {
+            if (string.IsNullOrWhiteSpace(markDownModel?.Body))
```
Why the service didn't validate the model ?

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Controllers/MarkDownController.cs:
```diff
@@ -0,0 +1,176 @@
+            }
+            catch (Exception ex)
+            {
+                return StatusCode((int)HttpStatusCode.InternalServerError, ex);
```
**Remark 3**
 Not needed, ASP.NET Core will generate an internal error 500 by default if you let the exception bubble.
It will also allow tools like Application Insights to "log" crashes in you app.

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Controllers/MarkDownController.cs:
```diff
@@ -0,0 +1,176 @@
+            }
+            catch (Exception ex)
+            {
+                return StatusCode((int)HttpStatusCode.InternalServerError, ex);
```
**Remark 3**

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Controllers/MarkDownController.cs:
```diff
@@ -0,0 +1,176 @@
+        [HttpGet]
+        [Route("api/markdowns/")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(List<MarkDownModel>))]
+        public async Task<IActionResult> GetAll()
```
**Remark 1&&2**

jgiacomini replied:
**Remark 5** 
I advice you to use as result type : `Task<List<MarkDownModel>>`  or `Task<ActionResult<MarkDownModel>> `. It will allow ASP.NET Core and swagger to map correctly your result to the good type without to maintain it with custom attribute

clemalex06 replied:
I could resolve all Remark 1&&2&&5 on the commit : 
https://github.com/clemalex06/BlazorMarkDown/pull/8/commits/614a92a5082bffb7dba45a98b6d1eef6dfdb552e

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Controllers/MarkDownController.cs:
```diff
@@ -0,0 +1,176 @@
+        [HttpPost]
+        [Route("api/markdown/")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(MarkDownModel))]
+        public async Task<IActionResult> Post([FromBody] MarkDownModel markDownModel)
```
**Remark 5**

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Controllers/MarkDownController.cs:
```diff
@@ -0,0 +1,176 @@
+        [HttpPut]
+        [Route("api/markdown/")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(MarkDownModel))]
+        public async Task<IActionResult> Put([FromBody] MarkDownModel markDownModel)
```
**Remark 1 && 2 && 5**

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Controllers/MarkDownController.cs:
```diff
@@ -0,0 +1,176 @@
+        [HttpGet]
+        [Route("api/markdown/readme/")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(MarkDownModel))]
+        public async Task<IActionResult> GetReadMe()
```
**Remark 1 && 2 && 5**

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Controllers/MarkDownController.cs:
```diff
@@ -0,0 +1,176 @@
+        [HttpGet]
+        [Route("api/markdown")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(MarkDownModel))]
+        public async Task<IActionResult> Get([FromQuery] long currentId)
```
**Remark 1 && 2 && 5**

jgiacomini reviewed (commented):
Hello,

I have a made a quick review.
I will try to dig a little bit more on the PR tommorrow morning.

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Controllers/MarkDownController.cs:
```diff
@@ -0,0 +1,176 @@
+                }
+                else
+                {
+                    throw new Exception("Unable to retrieve inserted document");
```
I don't understand the reason this exeption.
It's seem to be a bad pratice ?

clemalex06 replied:
Yes for this part really my bad, quite disapointed from my side about this beginner bad practice. 

I fixed it on https://github.com/clemalex06/BlazorMarkDown/pull/8/commits/614a92a5082bffb7dba45a98b6d1eef6dfdb552e

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Controllers/MarkDownController.cs:
```diff
@@ -0,0 +1,176 @@
+                }
+                else
+                {
+                    throw new Exception("Unable to retrieve updated document");
```
I don't understand the reason this exeption.
It's seem to be a bad pratice ?

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Controllers/AuthController.cs:
```diff
@@ -0,0 +1,88 @@
+        private string? CreateJWT(User user)
+        {
+            var secretkey = new SymmetricSecurityKey(System.Text.Encoding.UTF8.GetBytes("THIS IS THE SECRET KEY")); // NOTE: SAME KEY AS USED IN Program.cs FILE
+            var credentials = new SigningCredentials(secretkey, SecurityAlgorithms.HmacSha256);
```
Usually it's a good pratice to isolate this code it's own class.

clemalex06 replied:
I created a dedicated JWTHelper static class to manage Token generation in Backend Side.
https://github.com/clemalex06/BlazorMarkDown/pull/8/commits/614a92a5082bffb7dba45a98b6d1eef6dfdb552e

jgiacomini commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Controllers/MarkDownController.cs:
```diff
@@ -0,0 +1,176 @@
+                var document = await markDownService.GetDocument(currentId);
+
+                if (!string.IsNullOrWhiteSpace(document?.MarkDown))
+                {
```
It's a good pratice to return "NotFound" result when the entity is not found.
If you "hide" this behavior by creating empty markdown model you will have to write defensive on the UI side and it will be hard to debug and understand why you don't have a markdown value if it's not found

clemalex06 replied:
Agree with you, my bad to don't make it at the first implementation.
Fixed in : 
https://github.com/clemalex06/BlazorMarkDown/pull/8/files/7b17cdcde7ad6a7c2d66c22a58e512db49fe3da9..614a92a5082bffb7dba45a98b6d1eef6dfdb552e

jgiacomini commented:
The solution have 5 warnings it's not a good thing to have a small solution with 5 warnings by default : 
![image](https://user-images.githubusercontent.com/22032082/188571749-bacbb159-b77e-4acc-985d-3cb03e9d81a4.png)

Commit:
Commit to fix somes Pull Request remarks.

- Check any entry parameter send in AuthenticateUser, in UserService
- Fix Warnings on build + IntelliSense
- Fix Remark 1 to add Async suffix on async task on Controllers and service
- Fix Remark 2 : add CancellationToken as parameter to avoid useless processes.
- Fix Remark 5 on MarkDownController, change return Type from IActionResult to ActionResult<T>
- MarkDownController Remove Bad practice throx in a try block.
- MarkDownController, GetAsync and GetReadMeAsync, implement NotFound return response in case of unable to find the document.
- MarkDownComponent.razor, fix a detected bug.
- Create a dedicated JWTHelper static class to manage Token generation in Backend Side
- Adapt unit tests due to changes.

clemalex06 commented:
Hello @jgiacomini. First of all, really thanks a lot for time spent on the Pull Request.

I tried to make my best to solve somes of the remarks made, you can see all the changes on this commit : 
https://github.com/clemalex06/BlazorMarkDown/pull/8/commits/614a92a5082bffb7dba45a98b6d1eef6dfdb552e

clemalex06 commented:
I could fix the warnings, totally agree with you that is a bad practice to let warning. The lest there are, the best there is.

It seems Ok now after commit https://github.com/clemalex06/BlazorMarkDown/pull/8/commits/614a92a5082bffb7dba45a98b6d1eef6dfdb552e

![image](https://user-images.githubusercontent.com/9862193/189064166-f4c8f5a3-22f9-4577-88ad-ae21c31e94fd.png)

PierreBouillie commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Client/Pages/Login.razor:
```diff
@@ -0,0 +1,77 @@
+				}
+			}
+		}
+		catch
```
You should not have silent catch like that, it is a good practice to inform the user there was an error

PierreBouillie commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Services/MarkDowns/DocumentService.cs:
```diff
@@ -0,0 +1,71 @@
+            return document;
+        }
+
+        public async Task<Document?> InsertDocument(string markDown)
```
Since your method create the document the return should never be null

PierreBouillie commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Services/MarkDowns/DocumentService.cs:
```diff
@@ -0,0 +1,71 @@
+                document.MarkDown = markDown;
+                await ctx.SaveChangesAsync();
+            }
+            return document;
```
should it not throw a notfoundexception?

PierreBouillie commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Entities/User.cs:
```diff
@@ -0,0 +1,15 @@
+﻿namespace BlazorMarkDownAppJwt.Server.Entities
+{
+    public class User
```
Your user fields should not be nullable in the database since you are checking them 
            if (string.IsNullOrWhiteSpace(addedUser.Email)
                || string.IsNullOrWhiteSpace(addedUser.Password)
                || string.IsNullOrWhiteSpace(addedUser.FirstName)
                || string.IsNullOrWhiteSpace(addedUser.LastName))
                return null;

PierreBouillie commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Services/Users/UserService.cs:
```diff
@@ -0,0 +1,37 @@
+                return null;
+
+            var existingUser = await Ctx.Users.SingleOrDefaultAsync(u => u.Email == addedUser.Email);
+            if (existingUser != null) return null;
```
You should throw an exception in order to manage user exists

PierreBouillie commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Shared/AuthModel.cs:
```diff
@@ -0,0 +1,36 @@
+{
+	public class LoginResult
+	{
+		public string? message { get; set; }
```
FYI, string is e reference type and does not need to be have the "?"

StillShedy replied:
@PierreBouillie 
in .net 6 solution could be configured as Nullable enable
and in this case VS show you a warning if you dont have "?" after ref type
https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/tutorials/nullable-reference-types

StillShedy commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt.Server.UnitTests/Server/Controllers/AuthControllerTest.cs:
```diff
@@ -0,0 +1,193 @@
+
+        private AuthController Controller { get; set; }
+
+        private CancellationToken CancellationToken { get; set; }
```
https://docs.microsoft.com/en-us/dotnet/api/system.threading.cancellationtoken?view=net-6.0

if you need cancellation token you should take it from CancellationTokenSource

StillShedy commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Client/Components/MarkDownComponent.razor:
```diff
@@ -0,0 +1,233 @@
+		var jwt = await jsr.InvokeAsync<string>("localStorage.getItem", "user").ConfigureAwait(false);
+		if (!string.IsNullOrWhiteSpace(UserMail))
+		{
+			var dataArray = jwt.Split(';', 2);
```
magic symbols

StillShedy commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Client/Components/MarkDownComponent.razor:
```diff
@@ -0,0 +1,233 @@
+				await jsr.InvokeVoidAsync("localStorage.removeItem", "user").ConfigureAwait(false);
+				isLoading = false;
+				SetError("You are not authorized, please log in again");
+				nav.NavigateTo("/", true);
```
why do you use forceLoad everywhere?

StillShedy commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Client/Components/MarkDownComponent.razor:
```diff
@@ -0,0 +1,233 @@
+
+	private void SetError(string message)
+	{
+		markupString = (MarkupString)markDownModel.Html;
```
you can do this cast only once directly in razor

StillShedy commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Client/Pages/Index.razor:
```diff
@@ -0,0 +1,69 @@
+<h1>MarkDown List</h1>
+
+<p>Please choose one the markdown to view.</p>
+<SurveyPrompt Title="How BlazorMarkDownApp has been built from scratch?" />
```
Could you explain what is the component?

StillShedy commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Client/Pages/Register.razor:
```diff
@@ -0,0 +1,82 @@
+
+		}
+		catch
+		{
```
hiding exceptions

StillShedy commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Client/Shared/MainLayout.razor:
```diff
@@ -0,0 +1,57 @@
+
+	private async Task<string?> GetUserMail()
+	{
+		var jwt = await jsr.InvokeAsync<string>("localStorage.getItem", "user").ConfigureAwait(false);
```
why .ConfigureAwait(false)? not true? do we really need .ConfigureAwait ?

StillShedy commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Controllers/MarkDownController.cs:
```diff
@@ -0,0 +1,161 @@
+        [HttpPut]
+        [Route("api/markdown/")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(MarkDownModel))]
+        public async Task<ActionResult<MarkDownModel>> PutAsync([FromBody] MarkDownModel markDownModel, CancellationToken cancellationToken)
```
why in login endpoints you are returning Model and in this one you are using ActionResult?

StillShedy commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Program.cs:
```diff
@@ -0,0 +1,76 @@
+app.UseAuthorization(); // NOTE: line is newly addded, notice placement after UseRouting()
+
+
+app.MapRazorPages();
```
where do you use razor pages in your project?

StillShedy commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Services/MarkDowns/DocumentService.cs:
```diff
@@ -0,0 +1,74 @@
+            var path = Path.Combine(env.ContentRootPath, documentPath);
+            if (!Directory.Exists(path))
+                return null;
+            path = Path.Combine(path, "README.md");
```
why it's not comes from db? you made and editor tool, but made a doc in dedicated file
dubious product advertising :D

StillShedy commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Program.cs:
```diff
@@ -0,0 +1,76 @@
+// NOTE: end block
+
+
+builder.Services.AddControllersWithViews();
```
MVC + Razor Pages + Blazor do you use all of this tech?

StillShedy commented on BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Program.cs:
```diff
@@ -0,0 +1,76 @@
+{
+	app.UseExceptionHandler("/Error");
+	// The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
+	app.UseHsts();
```
do you know what it is?

Commit:
Some modifications before closing Peer Review
